modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -267,6 +280,54 @@ private String _buildFragmentHTML(
 		return sb.toString();
 	}
 
+	private JSONArray _getActionsJSONArray(
+			ObjectDefinition fdsViewObjectDefinition,
+			ObjectEntry fdsViewObjectEntry)
+		throws Exception {
+
+		Set<ObjectEntry> fdsActionObjectEntries = new TreeSet<>(
+			new ObjectEntryComparator(
+				ListUtil.toList(
+					ListUtil.fromString(
+						MapUtil.getString(
+							fdsViewObjectEntry.getProperties(),
+							"fdsActionsOrder"),
+						StringPool.COMMA),
+					Long::parseLong)));
+
+		fdsActionObjectEntries.addAll(
+			_getRelatedObjectEntries(
+				fdsViewObjectDefinition, fdsViewObjectEntry,
+				"fdsViewFDSActionRelationship"));
+
+		return JSONUtil.toJSONArray(
+			fdsActionObjectEntries,
+			(ObjectEntry fdsActionObjectEntry) -> {
+				Map<String, Object> properties =
+					fdsActionObjectEntry.getProperties();
+
+				return JSONUtil.put(
+					"data",
+					JSONUtil.put(
+						"confirmationMessage",
+						properties.get("confirmationMessage")
+					).put(
+						"status", properties.get("confirmationMessageType")
+					).put(
+						"title", properties.get("label")
```

Comment:
[opinionated] We're setting title of the confirmation modal here to be equal to `label`. We may want to decouple this and offer `confirmationMessageTitle` as an option in DSM.

Reply:
good point! Another alternative is to use a generic key kind of "Confirm", not sure what's best from UX perspective. Anyways we can go with this for now